python/ray/tune/session.py

```diff
@@ -14,10 +17,50 @@
     PlacementGroupSchedulingStrategy,
 )
 from ray.tune.error import TuneError
+from ray.tune.function_runner import _StatusReporter
 
 logger = logging.getLogger(__name__)
 
-_session = None
+_session: Optional[_StatusReporter] = None
+# V2 Session API.
+_session_v2: Optional["TuneSession"] = None
+
+
+class TuneSession(Session):
+    """Session client that function trainable can interact with."""
+
+    def __init__(self, status_reporter: _StatusReporter):
+        self._status_reporter = status_reporter
+
+    def report(self, metrics: Dict, checkpoint: Optional[Checkpoint] = None) -> None:
+        self._status_reporter.report(metrics, checkpoint)
+
+    @property
+    def loaded_checkpoint(self) -> Optional[Checkpoint]:
+        return self._status_reporter.loaded_checkpoint
+
+    @property
+    def trial_name(self) -> str:
+        """Trial name for the corresponding trial."""
+        return self._status_reporter.trial_name
+
+    @property
+    def trial_id(self) -> str:
+        """Trial id for the corresponding trial."""
+        return self._status_reporter.trial_id
+
+    @property
+    def logdir(self) -> str:
+        """Returns the directory that the session can safely work with.
+
+        If there are further distributed computation, child sessions can
+        then have separate dirs based off of this parent logdir."""
+        return self._status_reporter.logdir
+
+    @property
+    def trial_resources(self) -> PlacementGroupFactory:
+        """Trial resources for the corresponding trial."""
+        return self._status_reporter.trial_resources
```

Comment:
Can we unify the three sessions so they have the exact same API? This means the same set of method signatures, with no additional methods.